driver-core/src/main/java/com/datastax/driver/core/Clock.java

```diff
@@ -15,25 +15,169 @@
  */
 package com.datastax.driver.core;
 
+import com.sun.jna.Library;
+import com.sun.jna.Native;
+import com.sun.jna.Structure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
 /**
- * This interface allows us not to have a direct call to {@code System.currentTimeMillis()} for testing purposes
+ * A small abstraction around system clock that aims to provide microsecond precision with the best accuracy possible.
  */
 interface Clock {
+
     /**
-     * Returns the current time in milliseconds
+     * Returns the current time in microseconds
      *
-     * @return the difference, measured in milliseconds, between the current time and midnight, January 1, 1970 UTC.
-     * @see System#currentTimeMillis()
+     * @return the difference, measured in microsencods, between the current time and midnight, January 1, 1970 UTC.
      */
-    long currentTime();
+    long currentTimeMicroseconds();
```

Comment:
Changed the meaning of this method to return microseconds instead of milliseconds.